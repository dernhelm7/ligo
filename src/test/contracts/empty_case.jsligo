type Bar = {
  kind: 'Bar',
  value: int
};

type Baz = {
  kind: 'Baz'
}

type fooBarBaz =
| Bar
| Baz;

let main = (f : fooBarBaz) : int =>
  match(f, {
    Bar: (i: int) => i,
    Baz: () => -1
  })