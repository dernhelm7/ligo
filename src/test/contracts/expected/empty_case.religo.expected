type foo = Bar(int) | Baz;

let main = (f: foo): int => 
  switch(f){
  | Bar(i) => i
  | Baz => (-1)
  };
